### DISCUSSION TITLE
What is the proper way to update check box buttons?

### DISCUSSION THREAD

**Discussion by KourFrost** [py-pdf/pypdf#1112] [Q&A] [answered]
What is the proper way to update buttons or PDF Objects in general?

writer.update_page_form_field_values( ) method only updates the  "/V" attribute. How would you go about changing other attributes?

I'm pretty new at this, so I'm probably doing something wrong.

what I have tried:
```
import PyPDF2 
fileName = "example.pdf"
reader = PyPDF2.PdfReader(fileName)
w = PyPDF2.PdfWriter()
page = reader.pages[0]
w.add_page(page)

for annot in w.pages[0]["/Annots"]:
    obj = annot.get_object()
    if "/T" in obj.keys():
        if obj["/T"] == "Language 4 Check Box":
            print(obj)
            newValue = {"/AS": "/On", "/V": "/On"}
            obj.update(newValue)
            print(obj)

with open('updated_button2.pdf', "wb") as output_stream:
    w.write(output_stream)
```
[example.pdf](https://github.com/py-pdf/PyPDF2/files/9115079/example.pdf)

**Comment by MartinThoma** [comment, marked as answer]
Updating forms is still pretty cumbersome with PyPDF2. We have a couple of open issues: https://github.com/py-pdf/PyPDF2/labels/workflow-forms

One user suggested that we should work on FDF support. I haven't had the time to look into that.

**Reply by KourFrost** [reply]
Oh good to know it's not only me. I appreciate the fast response and PyPDF2 is a great tool. I'll take some time to see if I can add anything useful.
